addons/main/functions/fnc_getLauncherUnits.sqf

```diff
@@ -0,0 +1,63 @@
+#include "script_component.hpp"
+/*
+ * Author: ThomasAngel
+ * Gets all units from a group that have launchers.
+ * Things like AP launchers, flares etc will not count, -
+ * launcher must be capable of destroying at least some vehicle.
+ *
+ * Checking through the submunitions might improve mod compatibility,
+ * but only if the AI actually accounts for submunitions when evaluating -
+ * whether to engage or not. As of writing, this is unknown.
+ *
+ * Disposable launchers get mostly ignored, AI doesn't seem to even use them.
+ *
+ * Arguments:
+ * 0: Group to search in <GROUP, ARRAY<OBJECT>>
+ * 1: Flags to check Against <NUMBER>
+ * 2: Check through submunitions <BOOLEAN>
+ *
+ * Return Value:
+ * Array - units with launchers
+ *
+ * Example:
+ * [group bob] call lambs_main_fnc_getLauncherUnits;
+ *
+ * Public: Yes
+*/
+
+params [
+    ["_group", [], [grpNull, []]],
+    ["_flags", 896, [0]], // 896 == AI_AMMO_USAGE_FLAG_VEHICLE + AI_AMMO_USAGE_FLAG_AIR + AI_AMMO_USAGE_FLAG_ARMOUR
+    ["_checkSubmunition", false, [false]]
+];
+
+if (_group isEqualType grpNull) then {
+    _group = units _group;
+};
+private _suitableUnits = [];
+{
+    if ((secondaryWeapon _x) isEqualTo "") then {continue};
+    private _currentUnit = _x;
+
+    private _unitsMagazines = (magazines _currentUnit) + (secondaryWeaponMagazine _currentUnit);
```

Comment:
best include `handgunMagazine` as well, RHs for example has a grenade launcher for the pistol slot.

Reply:
it checks in L39 explicitly for secondaryWeapon so we can ignore that I would say

Reply:
actually hold up, secondary weapon in arma terms is always the launcher slot.

Reply:
yes that is correct we don't check for hand guns in this function

Reply:
> best include `handgunMagazine` as well, RHs for example has a grenade launcher for the pistol slot.

In the context that this function is used, I don't think grenade launchers should be detected. This function is used to detect anti-vehicle launchers to identify which units should be given orders to destroy vehicles. 40mm HEDP arguably could fit into this category, but I believe it's best for code simplicity to not bother with that. I think the AI will fire 40mm grenades into soft vehicle targets even without being explicitly told to do so.